src/kv/RocksDBStore.cc

```diff
@@ -697,34 +894,42 @@ void RocksDBStore::RocksDBTransactionImpl::merge(
   const string &k,
   const bufferlist &to_set_bl)
 {
+  bool is_default_cf;
+  rocksdb::ColumnFamilyHandle* cf_handle =
+    db->get_column_family_handle(prefix, is_default_cf);
+
   string key = combine_strings(prefix, k);
 
   // bufferlist::c_str() is non-constant, so we can't call c_str()
   if (to_set_bl.is_contiguous() && to_set_bl.length() > 0) {
-    bat.Merge(rocksdb::Slice(key),
-	       rocksdb::Slice(to_set_bl.buffers().front().c_str(),
-			    to_set_bl.length()));
+    bat.Merge(cf_handle, rocksdb::Slice(is_default_cf ? key : k),
+	       rocksdb::Slice(to_set_bl.buffers().front().c_str(), to_set_bl.length()));
```

Comment:
should we assert for we have only one buffer here? or should we do to_set_bl.c_cstr()?

Reply:
sorry, not get your idea. do you mean to_set_bl.buffers().front().?

Reply:
Seems the assumption is, we have only buffer to do c_str() here, either we can assert for it or we can use c_str() on the bufferlist which allocates contiguous memory in c_str() using rebuild_bl().

Reply:
I will check it. But I prefer we keep it as it is now, because my change doesn't touch this place. We may optimize it later if needed. Do you agree?

Reply:
Okay. To be sure, you can add an assert for the number of buffers to be one.

Reply:
agree. thanks.